swabbie-aws/src/main/kotlin/com/netflix/spinnaker/swabbie/aws/images/AmazonImageHandler.kt

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.swabbie.aws.images
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.kork.core.RetrySupport
+import com.netflix.spinnaker.swabbie.*
+import com.netflix.spinnaker.swabbie.aws.instances.AmazonInstance
+import com.netflix.spinnaker.swabbie.aws.launchconfigurations.AmazonLaunchConfiguration
+import com.netflix.spinnaker.swabbie.echo.Notifier
+import com.netflix.spinnaker.swabbie.exclusions.ResourceExclusionPolicy
+import com.netflix.spinnaker.swabbie.model.*
+import com.netflix.spinnaker.swabbie.orca.OrcaJob
+import com.netflix.spinnaker.swabbie.orca.OrcaService
+import com.netflix.spinnaker.swabbie.orca.OrchestrationRequest
+import kotlinx.coroutines.experimental.Deferred
+import kotlinx.coroutines.experimental.async
+import kotlinx.coroutines.experimental.runBlocking
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.util.*
+import kotlin.system.measureTimeMillis
+
+@Component
+class AmazonImageHandler(
+  registry: Registry,
+  clock: Clock,
+  notifier: Notifier,
+  resourceTrackingRepository: ResourceTrackingRepository,
+  resourceOwnerResolver: ResourceOwnerResolver<AmazonImage>,
+  exclusionPolicies: List<ResourceExclusionPolicy>,
+  applicationEventPublisher: ApplicationEventPublisher,
+  lockingService: Optional<LockingService>,
+  retrySupport: RetrySupport,
+  private val rules: List<Rule<AmazonImage>>,
+  private val imageProvider: ResourceProvider<AmazonImage>,
+  private val instanceProvider: ResourceProvider<AmazonInstance>,
+  private val launchConfigurationProvider: ResourceProvider<AmazonLaunchConfiguration>,
+  private val accountProvider: AccountProvider,
+  private val orcaService: OrcaService
+) : AbstractResourceTypeHandler<AmazonImage>(
+  registry,
+  clock,
+  rules,
+  resourceTrackingRepository,
+  exclusionPolicies,
+  resourceOwnerResolver,
+  notifier,
+  applicationEventPublisher,
+  lockingService,
+  retrySupport
+) {
+  override fun remove(markedResource: MarkedResource, workConfiguration: WorkConfiguration) {
```

Comment:
Worth calling this `delete` instead of remove for clarity all the way up? 
Are there any situations where something is not deleted when it is `remove`d?

(Not necessary to change in this PR but just a comment).

Reply:
Renaming to `delete` is a good idea.
On the following question, I suppose you are referring to when an orca job to delete a resource fails? In that case, the pattern to follow would be to poll Orca for a task until it's completed. That's legit and an action item to tackle before turning `delete` functionality on for real.

Reply:
Will have a PR to address ^^.